### PR TITLE
1. new environment class to represent environment and its state/actions

### PR DESCRIPTION
**Pull request by jmatosp**
First step to re factor:

New environment class witch will have Environment state and actions on it
ATM only if its immutable or not.

Next step:
 Move methods related to operations on Environment to this class

### CODE REVIEW

File: src/Environment.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Dotenv;
+
+/**
+ * Represents System Environment and actions on it
+ *
```
Comment (GrahamCampbell) [inline src/Environment.php]: cs

File: src/Environment.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Dotenv;
+
+/**
+ * Represents System Environment and actions on it
```
Comment (GrahamCampbell) [inline src/Environment.php]: cs

File: src/Environment.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+{
+    /**
+     * Are we immutable?
+     *
+     * @var bool
+     */
+    protected $immutable = true;
+
+    /**
+     * @param boolean $immutable
```
Comment (GrahamCampbell) [inline src/Environment.php]: cs

File: src/Environment.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+     *
+     * @return void
+     */
+    public function setImmutable($immutable)
+    {
+        $this->immutable = $immutable;
+    }
+
+    /**
+     * @return boolean
```
Comment (GrahamCampbell) [inline src/Environment.php]: cs

File: src/Loader.php [hunk 5]
```diff
@@ -22,11 +22,9 @@ class Loader
     protected $filePath;
 
     /**
-     * Are we immutable?
-     *
-     * @var bool
+     * @var Environment
```
Comment (GrahamCampbell) [inline src/Loader.php]: cs

**Comment by GrahamCampbell** [comment]
Im :-1: here tbh. This adds unneeded complexity.

**Comment by jmatosp** [comment]
The opposite, smaller code parts, easier to maintain, and OOP. probably more lines of code

**Comment by GrahamCampbell** [comment]
I think it's silly to add a class to encapsulate one variable. This only makes sense if we had more than one.

**Comment by GrahamCampbell** [comment]
Thanks anyway.

**Comment by jmatosp** [comment]
this was to break down  the previous PR in to smaller chunks, obviously  this class would end up bigger.

**Comment by GrahamCampbell** [comment]
> obviously this class would end up bigger.

What I meant was send minimal changes.

**Comment by GrahamCampbell** [comment]
I don't think anything really needs to be changed though, on reflection.

**Pull request closed without merging**